src/engine/controls/cuecontrol.cpp

```diff
@@ -766,7 +897,93 @@ void CueControl::hotcueGotoAndPlay(HotcueControl* pControl, double v) {
     }
 }
 
-void CueControl::hotcueActivate(HotcueControl* pControl, double v) {
+void CueControl::hotcueGotoAndLoop(HotcueControl* pControl, double v) {
+    if (!v) {
+        return;
+    }
+
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    CuePointer pCue(pControl->getCue());
+
+    // Need to unlock before emitting any signals to prevent deadlock.
+    lock.unlock();
+
+    if (!pCue) {
+        return;
+    }
+
+    double startPosition = pCue->getPosition();
+    if (startPosition == Cue::kNoPosition) {
+        return;
+    }
+
+    if (pCue->getType() == mixxx::CueType::Loop) {
+        hotcueGoto(pControl, v);
+        setCurrentSavedLoopControl(pControl);
+    } else if (pCue->getType() == mixxx::CueType::HotCue) {
+        hotcueGoto(pControl, v);
+        setBeatLoop(pCue->getPosition(), true);
+    } else {
+        return;
+    }
+
+    if (!isPlayingByPlayButton()) {
+        // cueGoto is processed asynchronously.
+        // avoid a wrong cue set if seek by cueGoto is still pending
+        m_bPreviewing = false;
+        m_iCurrentlyPreviewingHotcues = 0;
+        // don't move the cue point to the hot cue point in DENON mode
+        m_bypassCueSetByPlay = true;
+        m_pPlay->set(1.0);
+    }
+}
+
+void CueControl::hotcueLoopToggle(HotcueControl* pControl, double v) {
+    if (!v) {
+        return;
+    }
+
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    CuePointer pCue = pControl->getCue();
+
+    // Need to unlock before emitting any signals to prevent deadlock.
+    lock.unlock();
+
+    if (!pCue || pCue->getPosition() == Cue::kNoPosition) {
+        return;
+    }
+
+    switch (pCue->getType()) {
+    case mixxx::CueType::Loop: {
+        if (m_pCurrentSavedLoopControl != pControl) {
+            setCurrentSavedLoopControl(pControl);
+        } else {
+            bool loopActive = pControl->getStatus() == HotcueControl::Status::Active;
+            setLoop(pCue->getPosition(), pCue->getEndPosition(), !loopActive);
+        }
+    } break;
+    case mixxx::CueType::HotCue: {
+        setCurrentSavedLoopControl(nullptr);
```

Comment:
No, it also happens on serato controllers when you map CUE LOOP mode.

https://www.virtualdj.com/manuals/hardware/pioneer/ddjsx2/layout/pads/cueloop.html

Reply:
I'll add a comment.

Reply:
I am not sure if this is the same in Mixxx. 
For my understanding CUE LOOP will shift the key pads from the hot cues 1..8 to the SavedLoops 9..16 
This means the user on a Serato controller will always have two "Layers", unlike a Traktor user where both is mixed.

Reply:
Yes, but that is done in the controller mapping. Traktor controller have no CUE LOOP mode and just don't map these COs. I'll open a draft PR for my Roland DJ-505 mapping for demonstration.

Reply:
That's my point, what we need to consider and describe. 

We have the Serato user where the Controller applies an offset to the pads. 
And we have the Skins and Traktor controllers, which will mess that up. 

Since the cues are track properties, we cannot really avoid this mass-up. We can just consider and describe the edge cases carefully.
Maybe also a write a guide how a Serato controller should be mapped.

Reply:
I tried to document this here: https://github.com/mixxxdj/manual/pull/231/files#diff-77185901ba6fa67427de92b60abad8bce4b16f6a56265d5259d9e45c31173a17R1078

But yes, I should probably add that to the wiki as well and explain a bit more.

Reply:
I renamed `loop_toggle` to `cueloop` (that probably makes the the intention more clear) and added support for setting regular hotcues as requested elsewhere.